test/common/runtime/runtime_impl_test.cc

```diff
@@ -48,6 +48,23 @@ TEST(Random, SanityCheckOfUniquenessRandom) {
   EXPECT_EQ(num_of_results, results.size());
 }
 
+TEST(Random, SanityCheckOfStdLibRandom) {
+  Runtime::RandomGeneratorImpl random;
+
+  static const auto num_of_items = 100;
+  std::vector<uint64_t> v(num_of_items);
+  std::iota(v.begin(), v.end(), 0);
+
+  static const auto num_of_checks = 10000;
+  for (size_t i = 0; i < num_of_checks; ++i) {
+    const auto prev = v;
+    std::shuffle(v.begin(), v.end(), random);
+    EXPECT_EQ(v.size(), prev.size());
+    EXPECT_NE(v, prev);
+    EXPECT_FALSE(std::is_sorted(v.begin(), v.end()));
```

Comment:
This has obvious flakey test appearance, although the probability of a flake would presumably be very low (https://en.wikipedia.org/wiki/Bogosort comes to mind 😃 ). It's of course also *possible* that a perfectly correct random number generator results in the same shuffle occurring two times in a row, but this does seem extremely unlikely.

In practice, if you can execute this test a few times with some very high ``--runs_per_test` value like 10000+ and no problems maybe it's OK. Have you tried that?

Reply:
Yep, theoretically there is ~10000/100! chance of that test failing which is practically impossible.  Nevertheless I did run it overnight to confirm that this won't happen:
```
  Stats over 10000 runs: max = 20.2s, min = 5.7s, avg = 8.7s, dev = 2.8s
```